src/App.tsx

```diff
@@ -1,61 +1,132 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodoWithUser, User } from './types';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const initTodos: TodoWithUser[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [todoTitleError, setTodoTitleError] = useState(false);
+  const [todoUserId, setTodoUserId] = useState(0);
+  const [todoUserIdError, setTodoUserIdError] = useState(false);
+  const [todos, setTodos] = useState(initTodos);
+
+  function resetForm() {
+    setTodoUserId(0);
+    setTodoTitle('');
+  }
+
+  function handleOnAdd(event: React.FormEvent<HTMLFormElement>) {
```

Comment:
```suggestion
  function handleAddTodo(event: React.FormEvent<HTMLFormElement>) {
```
You don't need to combine `handle` and `on` prefixes. `handle` is for the handler functions, `on` is for props